src/test/nodestore/DatabaseShard_test.cpp

```diff
@@ -0,0 +1,894 @@
+                    time_t time0 = time(0);
+                    while (time(0) - time0 < shard_store_timeout && (boost::filesystem::exists(path, ec) || ec))
+                    {
+                        std::this_thread::yield();
+                        ec = {};
+                    }
+                }
+
+                BEAST_EXPECT(db->getCompleteShards() == bitmask2rangeset(i == 2 ? 2 : 0));
+            }
+
+            {
+                Env env {*this, test_config("", backendType, shard_dir.path())};
+                DatabaseShard *db = env.app().getShardStore();
+                BEAST_EXPECT(db);
+
+                test_data D(seedValue + i, 3);
+                if (!BEAST_EXPECT(D.make_ledgers(env)))
+                    return;
+
+                if (i == 2)
+                    wait_shard(db, 1);
+
+                BEAST_EXPECT(db->getCompleteShards() == bitmask2rangeset(i == 2 ? 2 : 0));
+
+                if (i == 2)
+                {
+                    for (std::uint32_t j = 0; j < ledgers_per_shard; ++j)
+                        check_ledger(D, db, D.ledgers_[j]);
+                }
+            }
+        }
+    }
+
+    void test_import (std::string const& backendType, std::uint64_t const seedValue)
+    {
+        using namespace test::jtx;
+
+        beast::temp_dir shard_dir;
+        {
+            beast::temp_dir node_dir;
+            Env env {*this, test_config("import", backendType, shard_dir.path(), node_dir.path())};
+            DatabaseShard *db = env.app().getShardStore();
+            Database &ndb = env.app().getNodeStore();
+            BEAST_EXPECT(db);
+
+            test_data D(seedValue, 4, 4);
+            if (!BEAST_EXPECT(D.make_ledgers(env)))
+                return;
+
+            for (std::uint32_t i = 0; i < 4 * ledgers_per_shard; ++i)
+                BEAST_EXPECT(save_ledger(&ndb, D.ledgers_[i]));
+
+            BEAST_EXPECT(db->getCompleteShards() == bitmask2rangeset(0));
+            db->import(ndb);
+            for (std::uint32_t i = 1; i <= 4; ++i)
+                wait_shard(db, i);
+            BEAST_EXPECT(db->getCompleteShards() == bitmask2rangeset(0x1E));
+        }
+        {
+            Env env {*this, test_config("", backendType, shard_dir.path())};
+            DatabaseShard *db = env.app().getShardStore();
+            BEAST_EXPECT(db);
+
+            test_data D(seedValue, 4, 4);
+            if (!BEAST_EXPECT(D.make_ledgers(env)))
+                return;
+
+            for (std::uint32_t i = 1; i <= 4; ++i)
+                wait_shard(db, i);
+
+            BEAST_EXPECT(db->getCompleteShards() == bitmask2rangeset(0x1E));
+
+            for (std::uint32_t i = 0; i < 4 * ledgers_per_shard; ++i)
+                check_ledger(D, db, D.ledgers_[i]);
+        }
+    }
+
+    void test_all (std::string const& backendType)
+    {
+        std::uint64_t const seedValue = 51;
+        test_standalone(backendType);
+        test_create_shard(backendType, seedValue);
```

Comment:
There are some inconsistencies in the repo but the majority of method names seem to be camelCase as well.

Reply:
Done